config/src/main/java/com/typesafe/config/impl/ConfigDocumentParser.java

```diff
@@ -381,6 +395,7 @@ private void parseInclude(ArrayList<AbstractConfigNode> children) {
             } else {
                 throw parseError("include keyword is not followed by a quoted string, but by: " + t);
             }
+            return new ConfigNodeInclude(children);
```

Comment:
the list of include types (url, file, classpath) are parsed in this parseInclude method, but then it looks like we toss the result, which makes me expect that the same switch on url/file/classpath is going to be elsewhere in the code - I think only one place in the code should contain the four strings (url, file, classpath, error) and then all other places should have something pre-parsed to work with. further down maybe some ideas on where to consolidate